src/Support/Traits/HasDirtyState.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Statamic\Support\Traits;
+
+trait HasDirtyState
+{
+    /**
+     * Is the item or property on the item dirty?
+     *
+     * @param  null|string|array  $properties
+     * @return bool
+     */
+    public function isDirty($properties = null)
```

Comment:
`isDirty` seems to be returning `true`, even without me making any changes:

```php
use Statamic\Facades\Entry;

$entry = Entry::find('home');
$entry->isDirty();
```

This is only happening for entries & terms. I wonder if it might be related to the `json_encode` stuff on line 19. 🤔

Reply:
Just for niceties, can we add a `bool` type hint to this method?

Reply:
Done

Reply:
> This is only happening for entries & terms. I wonder if it might be related to the json_encode stuff on line 19

Think it was due to updated_at - I have it ignoring that now

Reply:
Nope, seems syncOriginal() isnt being called now, I'll need to dig into why

Reply:
Ah actually it is - can I check, did you do a `cache:clear`, the original state gets stored in the stache

Reply:
Pulled down your latest commits & ran `php artisan cache:clear` and it's working fine now.